test/test_rest_vol.c

```diff
@@ -8711,23 +8720,159 @@ test_write_dataset_data_verification(void)
     }
 
     /* Write to first two rows of dataset */
+    mdims[0]  = dims[1] * 2;
+    start[0]  = 0;
+    stride[0] = 1;
+    count[0]  = dims[1] * 2;
+    block[0]  = 1;
+    if ((mspace_id = H5Screate_simple(1, mdims, NULL)) < 0)
+        TEST_ERROR
+    if (H5Sselect_hyperslab(mspace_id, H5S_SELECT_SET, start, stride, count, block) < 0)
```

Comment:
Note that selecting a hyperslab in the memory space here isn't really necessary since it's just a contiguous block across all `dims[1] * 2` elements in the memory space that is created, but it's still ok to do so here.

Reply:
Ah, good to know this. In our code, we decided to always explicitly select a hyperslab to avoid the confusion which points have been selected.